xbmc/ServiceBroker.h

```diff
@@ -131,6 +132,10 @@ class CServiceBroker
   static void RegisterDecoderFilterManager(CDecoderFilterManager* manager);
   static CDecoderFilterManager* GetDecoderFilterManager();
 
+  static std::shared_ptr<CCPUInfo> GetCPUInfo();
```

Comment:
Why make this a shared pointer? Does some component need to use the cpu info for an extended period of time so that it might vanish during use? Or is this to prevent races on shutdown?

Reply:
I haven't tested this yet to be honest. I think originally I was just worried about multiple locations requesting the pointer at once.